scan-payment/src/main/java/com/getbouncer/scan/payment/ml/TextDetector.kt

```diff
@@ -0,0 +1,370 @@
+package com.getbouncer.scan.payment.ml
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.RectF
+import android.util.Log
+import android.util.Size
+import com.getbouncer.scan.framework.Loader
+import com.getbouncer.scan.framework.ResourceLoader
+import com.getbouncer.scan.framework.ml.TFLAnalyzerFactory
+import com.getbouncer.scan.framework.ml.TensorFlowLiteAnalyzer
+import com.getbouncer.scan.framework.ml.ssd.hardNonMaximumSuppression
+import com.getbouncer.scan.framework.ml.ssd.rectForm
+import com.getbouncer.scan.payment.R
+import com.getbouncer.scan.payment.hasOpenGl31
+import com.getbouncer.scan.payment.ml.ssd.DetectionBox
+import com.getbouncer.scan.payment.ml.yolo.processYoloLayer
+import com.getbouncer.scan.payment.scale
+import com.getbouncer.scan.payment.toRGBByteBuffer
+import org.tensorflow.lite.Interpreter
+import java.io.FileNotFoundException
+import java.nio.ByteBuffer
+import kotlin.math.abs
+import kotlin.math.floor
+import kotlin.math.ln
+import kotlin.math.max
+import kotlin.math.pow
+
+private val TRAINED_IMAGE_SIZE = Size(416, 416)
+
+private const val YOLO_POST_PROCESS_CONFIDENCE_THRESHOLD = 0.5f
+private val YOLO_ANCHORS = arrayOf(
+    arrayOf(
+        Pair(81, 82),
+        Pair(135, 169),
+        Pair(344, 319)
+    ),
+    arrayOf(
+        Pair(10, 14),
+        Pair(23, 27),
+        Pair(37, 58)
+    )
+)
+
+/**
+ * Model returns the following 4 classes:
+ * 0. MM/YY Expiration Date
+ * 1: Mostly letters - At most 1 non-character and more letters than other characters
+ * 2: Mostly numbers - At most 2 non-numbers and more numbers than other characters
+ * 3: Mixed - mix of numbers and characters (any remaining cases)
+ **/
+private const val NUM_CLASS = 4
+
+private val LAYER_1_SIZE = Size(13, 13)
+private val LAYER_2_SIZE = Size(26, 26)
+
+// A YOLO3 constant derived from NUM_CLASS
+private const val DIM_Z = (NUM_CLASS + 5) * 3
+
+class TextDetector private constructor(interpreter: Interpreter) :
+    TensorFlowLiteAnalyzer<TextDetector.Input, Array<ByteBuffer>,
+        TextDetector.Prediction,
+        Map<Int, Array<Array<Array<FloatArray>>>>>(interpreter) {
+
+    data class Input(val objDetectionImage: Bitmap)
```

Comment:
make this a full screen image that this analyzer knows how to crop. this will provide consistency across the analyzers.